restreamer/restreamer/main.py

```diff
@@ -493,23 +480,43 @@ def get_frame(channel, quality):
 
 @app.route('/thumbnail/<channel>/<quality>.png')
 @request_stats
-@has_path_args
 def get_thumbnail(channel, quality):
 	"""
 	Returns a PNG image which is a preview of how a thumbnail will be generated.
-	Params:
-		timestamp: Required. The frame to use as the thumbnail image.
-			Must be in ISO 8601 format (ie. yyyy-mm-ddTHH:MM:SS) and UTC.
-		template: Required. The template name to use.
-			Must be one of the template names (without file extension) as returned
-			by GET /files/thumbnail_templates
 	"""
-	template_name = request.args['template']
-	template_path = os.path.join(app.static_folder, "thumbnail_templates", f"{template_name}.png")
-	if not os.path.exists(template_path):
-		return "No such template", 404
 
-	timestamp = dateutil.parse_utc_only(request.args['timestamp'])
+	template_params = request.json
```

Comment:
This is a GET request, `request.json` shouldn't be used as it reads json from the request body, and while GET requests with a body may technically work most of the time, it's officially disallowed by the standard and not everything will support it.

I think the cleanest option here would be to have two versions of this function:
- A `GET` endpoint which uses query params to take a timestamp, template name and crop metadata
- A `POST` endpoint which uses query params to take a timestamp and crop metadata, but gets the template image from the request body

This also saves having to base64 the image template when sending it, which makes it a bit nicer to use from `curl` etc.

So something like:
```
@app.route("/thumbnail/<channel>/<quality>.png")
@request_stats
@has_path_args
def get_thumbnail_named_template(channel, quality):
    template = get_template_from_db(request.args["template"])
    return get_thumbnail(channel, quality, template)

@app.route("/thumbnail/<channel>/<quality>.png")
@request_stats
@has_path_args
def get_thumbnail_uploaded_template(channel, quality):
    template = request.body
    return get_thumbnail(channel, quality, template)
```

Reply:
I assume the second one should start `@app.route("/thumbnail/<channel>/<quality>.png" , methods=['POST'])` otherwise they have the same route.

Reply:
yes, whoops.